Sources/Build/llbuild.swift

```diff
@@ -13,6 +13,21 @@ import Utility
 import PackageModel
 import PackageGraph
 
+/// An enum representing the different build options.
+public enum BuildOption {
+    /// Represents the option to build all targets and products (excluding test targets).
+    case allExcludingTests
+
+    /// Represents the option to build all targets and products (including test targets).
+    case allIncludingTests
+
+    /// Represents the option to build a specific target.
+    case target(ResolvedTarget)
+
+    /// Represents the option to build a specific product.
+    case product(ResolvedProduct)
```

Comment:
Fair enough. How about the argument that it makes more difficult to make sure only one place in the code resolves those names? It seems to me like both the tool and the build process needs them resolved.